Xamarin.Forms.Core.UnitTests/NativeBindingTests.cs

```diff
@@ -0,0 +1,487 @@
+			}
+		}
+
+	}
+
+	class MockNativeViewWrapper : View
+	{
+		public MockNativeView NativeView { get; }
+
+		public MockNativeViewWrapper(MockNativeView nativeView)
+		{
+			NativeView = nativeView;
+			nativeView.TransferbindablePropertiesToWrapper(this);
+		}
+
+		protected override void OnBindingContextChanged()
+		{
+			NativeView.SetBindingContext(BindingContext, nv => nv.SubViews);
+			base.OnBindingContextChanged();
+		}
+
+	}
+
+	public class MockNativeColor
+	{
+
+		public MockNativeColor(Color color)
+		{
+			FormsColor = color;
+		}
+
+		public Color FormsColor
+		{
+			get;
+			set;
+		}
+	}
+
+	public static class MockNativeViewExtensions
+	{
+		public static View ToView(this MockNativeView nativeView)
+		{
+			return new MockNativeViewWrapper(nativeView);
+		}
+
+		public static void SetBinding(this MockNativeView target, string targetProperty, BindingBase binding, string updateSourceEventName = null)
+		{
+			NativeBindingHelpers.SetBinding(target, targetProperty, binding, updateSourceEventName);
+		}
+
+		internal static void SetBinding(this MockNativeView target, string targetProperty, BindingBase binding, INotifyPropertyChanged propertyChanged)
+		{
+			NativeBindingHelpers.SetBinding(target, targetProperty, binding, propertyChanged);
+		}
+
+		public static void SetBinding(this MockNativeView target, BindableProperty targetProperty, BindingBase binding)
+		{
+			NativeBindingHelpers.SetBinding(target, targetProperty, binding);
+		}
+
+		public static void SetValue(this MockNativeView target, BindableProperty targetProperty, object value)
+		{
+			NativeBindingHelpers.SetValue(target, targetProperty, value);
+		}
+
+		public static void SetBindingContext(this MockNativeView target, object bindingContext, Func<MockNativeView, IEnumerable<MockNativeView>> getChild = null)
+		{
+			NativeBindingHelpers.SetBindingContext(target, bindingContext, getChild);
+		}
+
+		internal static void TransferbindablePropertiesToWrapper(this MockNativeView target, MockNativeViewWrapper wrapper)
+		{
+			NativeBindingHelpers.TransferbindablePropertiesToWrapper(target, wrapper);
+		}
+	}
+
+	class MockCustomColorConverter : IValueConverter
+	{
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+		{
+			if (value is Color)
+				return new MockNativeColor((Color)value);
+			return value;
+		}
+
+		public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+		{
+			if (value is MockNativeColor)
+				return ((MockNativeColor)value).FormsColor;
+			return value;
+		}
+	}
+
+	class MockINPC : INotifyPropertyChanged
+	{
+		public void FireINPC(object sender, string propertyName)
+		{
+			PropertyChanged?.Invoke(sender, new PropertyChangedEventArgs(propertyName));
+		}
+
+		public event PropertyChangedEventHandler PropertyChanged;
+	}
+
+	class MockVMForNativeBinding : INotifyPropertyChanged
+	{
+		string fFoo;
+		public string FFoo {
+			get { return fFoo; }
+			set {
```

Comment:
Okay, but the curly brace placement should at least be the same within the same check-in.